planner/core/plan_cost_ver2.go

```diff
@@ -541,71 +541,119 @@ func (p *BatchPointGetPlan) getPlanCostVer2(taskType property.TaskType, option *
-	hashMemCost := buildRows * buildRowSize * memFactor
-	hashBuildCost := buildRows * float64(len(keys)) * cpuFactor
+	hashKeyCost := buildRows * float64(len(keys)) * cpuFactor.Value
+	hashMemCost := buildRows * buildRowSize * memFactor.Value
+	hashBuildCost := buildRows * float64(len(keys)) * cpuFactor.Value
 	return hashKeyCost + hashMemCost + hashBuildCost
 }
 
-func hashProbeCostVer2(probeRows float64, keys []expression.Expression, cpuFactor float64) float64 {
+func hashProbeCostVer2(probeRows float64, keys []expression.Expression, cpuFactor costVer2Factor) float64 {
 	// TODO: 1) consider types of keys, 2) dedicated factor for build-probe hash table
-	hashKeyCost := probeRows * float64(len(keys)) * cpuFactor
-	hashProbeCost := probeRows * float64(len(keys)) * cpuFactor
+	hashKeyCost := probeRows * float64(len(keys)) * cpuFactor.Value
+	hashProbeCost := probeRows * float64(len(keys)) * cpuFactor.Value
 	return hashKeyCost + hashProbeCost
 }
 
-func getTaskCPUFactor(p PhysicalPlan, taskType property.TaskType) float64 {
+type costVer2Factor struct {
+	Name  string
+	Value float64
+}
+
+// In Cost Ver2, we hide cost factors from users and deprecate SQL variables like `tidb_opt_scan_factor`.
+type costVer2Factors struct {
+	TiDBTempScan  costVer2Factor // TiDB temporary table scan
+	TiKVScan      costVer2Factor // per byte
+	TiKVDescScan  costVer2Factor // per byte
+	TiFlashScan   costVer2Factor // per byte
+	TiDBCPU       costVer2Factor // per column or expression
+	TiKVCPU       costVer2Factor // per column or expression
+	TiFlashCPU    costVer2Factor // per column or expression
+	TiDB2KVNet    costVer2Factor // per byte
+	TiDB2FlashNet costVer2Factor // per byte
+	TiFlashMPPNet costVer2Factor // per byte
+	TiDBMem       costVer2Factor // per byte
+	TiKVMem       costVer2Factor // per byte
+	TiFlashMem    costVer2Factor // per byte
+	TiDBDisk      costVer2Factor // per byte
+	TiDBRequest   costVer2Factor // per net request
+}
+
+var defaultVer2Factors = costVer2Factors{
+	TiDBTempScan:  costVer2Factor{"tidb_temp_table_scan_factor", 0},
+	TiKVScan:      costVer2Factor{"tikv_scan_factor", 100},
+	TiKVDescScan:  costVer2Factor{"tikv_desc_scan_factor", 150},
+	TiFlashScan:   costVer2Factor{"tiflash_scan_factor", 5},
+	TiDBCPU:       costVer2Factor{"tidb_cpu_factor", 30},
+	TiKVCPU:       costVer2Factor{"tikv_cpu_factor", 30},
+	TiFlashCPU:    costVer2Factor{"tiflash_cpu_factor", 5},
+	TiDB2KVNet:    costVer2Factor{"tidb_kv_net_factor", 8},
+	TiDB2FlashNet: costVer2Factor{"tidb_flash_net_factor", 4},
+	TiFlashMPPNet: costVer2Factor{"tiflash_mpp_net_factor", 4},
+	TiDBMem:       costVer2Factor{"tidb_mem_factor", 1},
+	TiKVMem:       costVer2Factor{"tikv_mem_factor", 1},
+	TiFlashMem:    costVer2Factor{"tiflash_mem_factor", 1},
+	TiDBDisk:      costVer2Factor{"tidb_disk_factor", 1000},
+	TiDBRequest:   costVer2Factor{"tidb_request_factor", 9500000},
+}
+
+func getTaskCPUFactor(p PhysicalPlan, taskType property.TaskType) costVer2Factor {
```

Comment:
Do we need to name it and the methods below something like `getTaskCPUFactor`? There are methods with very similar names in `plan_cost.go`.

Reply:
OK, I renamed them to `getXXXFactorVer2` to indicate they are only used by model ver2.